clusterman/aws/aws_resource_group.py

```diff
@@ -60,6 +60,14 @@ class AWSResourceGroup(ResourceGroup, metaclass=ABCMeta):
     def __init__(self, group_id: str, **kwargs: Any) -> None:
         self.group_id = group_id
 
+        # Resource Groups are reloaded on every autoscaling run, so we just query
+        # AWS data once and store them so we don't run into AWS request limits
+        #
+        # This is expected to populate self.instance_ids, which has to be done _before_
+        # we populate the instances by market
+        self._reload_resource_group()
+        self._instances_by_market = self._get_instances_by_market()
```

Comment:
nit: we're not really *re*loading them since we only load them once on initialization, yeah?  And why not put the `instance_by_market` stuff in that same method?

Reply:
Well, you're _technically_ correct (the best kind of correct) that we're not re-loading them but everything above this _is_ getting reloaded (e.g. the pool manager), so I sorta feel like I'd like to keep the terminology consistent... and from a "what's actually happening" perspective, the resource groups are getting reloaded.  It just so happens that we "reload" them by re-creating them.

The `_reload_resource_group` method is customized per resource-group type, so I don't think it makes sense to make people have to remember to include that everywhere.